python/lsst/ap/association/assoc_afw_utils.py

```diff
@@ -0,0 +1,279 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+from collections import OrderedDict as oDict
+
+import numpy as np
+
+import lsst.afw.table as afwTable
+
+__all__ = ["make_minimal_dia_object_schema",
+           "make_minimal_dia_source_schema",
+           "ccdVisitSchema"]
+
+
+"""Defines afw schemas and conversions for use in ap_association tasks.
+
+Until a more finalized interface between the prompt products database (PPD) can
+be established, we put many utility functions for converting `lsst.afw.table` and
+`lsst.afw.image` objects to a PPD. This includes both mapping schemas between
+different catalogs and the DB.
+"""
+
+
+def make_minimal_dia_object_schema(filter_names=[]):
+    """Define and create the minimal schema required for a DIAObject.
+
+    Parameters
+    ----------
+    filter_names : `list` of `str`s
+        Names of the filters expect and compute means for.
+
+    Return
+    ------
+    schema : `lsst.afw.table.Schema`
+        Minimal schema for DIAObjects.
+    """
+    schema = afwTable.SourceTable.makeMinimalSchema()
+    # For the MVP/S we currently only care about the position though
+    # in the future we will add summary computations for fluxes etc.
+    # as well as their errors.
+
+    # In the future we would like to store a covariance of the coordinate.
+    # This functionality is not defined in currently in the stack, so we will
+    # hold off until it is implemented. This is to be addressed in DM-7101.
+    schema.addField("pixelId", type='L')
+    schema.addField("nDiaSources", type='L')
+    for filter_name in filter_names:
+        schema.addField("psFluxMean_%s" % filter_name, type='D')
+        schema.addField("psFluxMeanErr_%s" % filter_name, type='D')
+        schema.addField("psFluxSigma_%s" % filter_name, type='D')
+    return schema
+
+
+def make_minimal_dia_source_schema():
+    """ Define and create the minimal schema required for a DIASource.
+
+    Return
+    ------
+    schema : `lsst.afw.table.Schema`
+        Minimal schema for DIAObjects.
+    """
+    schema = afwTable.SourceTable.makeMinimalSchema()
+    schema.addField("diaObjectId", type='L')
+    schema.addField("ccdVisitId", type='L')
+    schema.addField("psFlux", type='D')
+    schema.addField("psFluxErr", type='D')
+    schema.addField("filterName", type='String', size=10)
+    schema.addField("filterId", type='L')
+    return schema
+
+
+def ccdVisitSchema():
+    """Define the schema for the CcdVisit table.
```

Comment:
Document whether this is for an afw table, or for an sql table?

It should probably be called `getCcdVisitSchema()`, or `getCcdVisitSqlSchema()` if it's for SQL and not AFW.